### ISSUE TITLE
Rule Change: (Can the curly rule support that all control structures are required to be block statements, but a single-line statement can be a block structure or not.)

### ISSUE THREAD

**Issue by NieZhuZhu** [eslint/eslint#18062]
### What rule do you want to change?

curly

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
function foo() {
    if (globalEnv.isInVolc) return true;
    if (globalEnv.isVeStackHybrid) return true;
   
    if (globalEnv.aaaa) {
      return true;
    }
   
    return false;
  }
```

### What does the rule currently do for this code?

```javascript

/*eslint curly: ["error", "multi-or-nest", "consistent"]*/
 function enableProject() {
    if (globalEnv.isInVolc) return true;
    if (globalEnv.isVeStackHybrid) return true;
    // Unnecessary { after 'if' condition.  ([curly](https://eslint.org/docs/rules/curly))
    if (globalEnv.aaaa) {
      return true;
    }

   if (globalEnv.bbb) {
      console.log(bbb)
      return true;
    }
   
    return false;
}


```

[playgroud](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgY3VybHk6IFtcImVycm9yXCIsIFwibXVsdGktb3ItbmVzdFwiLCBcImNvbnNpc3RlbnRcIl0qL1xuIGZ1bmN0aW9uIGVuYWJsZVByb2plY3QoKSB7XG4gICAgaWYgKGdsb2JhbEVudi5pc0luVm9sYykgcmV0dXJuIHRydWU7XG4gICAgaWYgKGdsb2JhbEVudi5pc1ZlU3RhY2tIeWJyaWQpIHJldHVybiB0cnVlO1xuICAgXG4gICAgaWYgKGdsb2JhbEVudi5hYWFhKSB7XG4gICAgICByZXR1cm4gdHJ1ZTtcbiAgICB9XG5cbiAgIGlmIChnbG9iYWxFbnYuYmJiKSB7XG4gICAgICBjb25zb2xlLmxvZyhiYmIpXG4gICAgICByZXR1cm4gdHJ1ZTtcbiAgICB9XG4gICBcbiAgICByZXR1cm4gZmFsc2U7XG59XG4iLCJvcHRpb25zIjp7ImVudiI6e30sInJ1bGVzIjp7fSwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9LCJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUifX19)

### What will the rule do after it's changed?

There are no options on this rule that can meet my needs. I want the curly rule to support that all control structures are required to be block statements, but a single-line statement can be a block structure or not.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by Rec0iL99** [comment]
Hi @NieZhuZhu, thanks for the issue. We are not adding any new options to stylistic rules due to the [policy change](https://github.com/eslint/eslint#stylistic-rule-updates). You can always copy the core rule into a custom rule and add more options.
